### ISSUE TITLE
[11.x] Calling `Middleware::trustHosts()` with arguments breaks entire application

### ISSUE THREAD

**Issue by axlon** [laravel/framework#50845]
### Laravel Version

11.1

### PHP Version

11.1.1

### Database Driver & Version

n/a

### Description

The new bootstrapping process breaks when calling `$middleware->trustHosts(...)` with any arguments while `$subdomains` is set to true. This is because doing so will cause the middleware to prepare its list of trusted domains, hen the middleware does this it calls the `config()` helper which hasn't yet been resolved at this point.

PS. I'm open to fixing this myself; I think the easiest way to fix this is to defer the configuration of the middleware to a booted callback, that way we can be sure that the config has been resolved.

PPS. If the above is agreeable, I'd like to also add the ability to call `$middleware->trustHosts()` with a callback. I initially ran into the problem described in this issue in my own code, because I was trying to register additional domains from my config (which cannot be done currently as I can only pass an array of already resolved domains)

### Steps To Reproduce

I've made an example repo [here](https://github.com/axlon/laravel-trust-hosts-issue/commits/main/) which shows the problem.

Basically:

```php
$middleware->trustHosts(); // Works
$middleware->trustHosts(subdomains: false); // Kinda works... still registers subdomains
$middleware->trustHosts([...], subdomains: true); // Error! Nothing works
$middleware->trustHosts([...], subdomains: false); // Works
```

**Comment by RudahXimenes** [comment]
I second this issue.

Is there any alternative to set TrustHosts while it is not fixed?

**Comment by gzwan** [comment]
Is there any alternative to set TrustHosts while it is not fixed?

**Comment by JurianArie** [comment]
@RudahXimenes @gzwan

You can copy the middleware from 10.x https://github.com/laravel/laravel/blob/10.x/app/Http/Middleware/TrustHosts.php and replace the middleware in you `bootstrap/app.php`:
```php
->withMiddleware(function (\Illuminate\Foundation\Configuration\Middleware $middleware) {
    $middleware->replace(\Illuminate\Http\Middleware\TrustHosts::class, \App\Http\Middleware\TrustHosts::class);
})
```

**Comment by RudahXimenes** [comment]
> @RudahXimenes @gzwan
> 
> You can copy the middleware from 10.x https://github.com/laravel/laravel/blob/10.x/app/Http/Middleware/TrustHosts.php and replace the middleware in you `bootstrap/app.php`:
> 
> ```
> ->withMiddleware(function (\Illuminate\Foundation\Configuration\Middleware $middleware) {
>     $middleware->replace(\Illuminate\Http\Middleware\TrustHosts::class, \App\Http\Middleware\TrustHosts::class);
> })
> ```

Thank you, buddy

I'll take a look and implement it this way!
